libs/ardour/audioregion.cc

```diff
@@ -1243,6 +1243,20 @@ AudioRegion::fade_out_is_default () const
 	return _fade_out->size() == 2 && _fade_out->when(true) == 0 && _fade_out->when(false) == 64;
 }
 
+samplecnt_t
+AudioRegion::fade_in_length ()
+{
+	samplecnt_t fade_in_length = (samplecnt_t) _fade_in->when(false);
```

Comment:
Why use a temporary variable instead of ` return _fade_in->when(false)`?

ControlList::when (false) returns the position of the last point, which is only the length if  `ControlList::when(true) == 0`.

Either name the function `fade_in_end_position()`, or subtract   `_fader_in->when(true)`.

Reply:
> Why use a temporary variable instead of return _fade_in->when(false)?

TBH, I can't remember, indeed it's useless.

I didn't know it is possible that fade in could start before or after the region start. Is it really possible ? I don't see how to do it in the GUI. The script doesn't manage this possibility.